main.cpp

```diff
@@ -39,8 +39,10 @@ struct Compare                                //4
     {
         if(a != nullptr && b != nullptr)
         {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
+            T &aa = *a;
+            T &bb = *b;
+            if( aa.value < bb.value ) return &aa;
+            if( aa.value > bb.value ) return &bb;
```

Comment:
You can convert the function arguments to references.  You can't convert the return type.
- [ ] revise